File: code/gdoc_frontend/src/service/fileService.js
import {apiUrl} from "../util/constants";
import {postRequest_query} from "../util/ajax";

export const create = (data,callback) => {
    const url = `${apiUrl}/create`;
    postRequest_query(url, data, callback);
};

export const mkdir = (data,callback) => {
    const url = `${apiUrl}/mkdir`;
    postRequest_query(url, data, callback);
};

export const readdir = (data,callback) => {
    const url = `${apiUrl}/dirinfo`;
    postRequest_query(url, data, callback);
};

export const deletefile = (data,callback) => {
    const url = `${apiUrl}/delete`;
    postRequest_query(url, data, callback);
};

export const recoverfile = (data,callback) => {
    const url = `${apiUrl}/recover`;
    postRequest_query(url, data, callback);
};

export const renamefile = (data,callback) => {
    const url = `${apiUrl}/rename`;
    postRequest_query(url, data, callback);
};

export const sharefile = (data,callback) => {
    const url = `${apiUrl}/share`;
    postRequest_query(url, data, callback);
};

export const read = (data,callback) => {
    const url = `${apiUrl}/read`;
    postRequest_query(url, data, callback);
};

export const getfileinfo = (data,callback) => {
    const url = `${apiUrl}/fileinfo`;
    postRequest_query(url, data, callback);
};

export const getloginfo = (data,callback) => {
    const url = `${apiUrl}/loginfo`;
    postRequest_query(url, data, callback);
};

File: code/gdoc_backend/src/main/java/gdoc/service/FileService.java
package gdoc.service;


import org.springframework.web.bind.annotation.RequestParam;

import java.io.IOException;
import java.util.List;

public interface FileService {

    public String create(String username, String filepath);
    public String share(String username,String shareusername, String filepath);
    public String fileinfo(String username, String filepath);
    public String[] loginfo(String username,String owner, String filepath);
    public String mkdir(String username,String dirpath);
    public String dirinfo(String username,String dirpath);
    public String rename(String username,String oldpath,String newpath);
    public String delete(String username, String filename);
    public String recover(String username, String filename);
    public String[] read(String username, String owner, String filepath);
    public String write(String username, String owner,String filepath,int row,int column,String content);
    public String trylock(String username, String owner,String filepath,int row,int column);
    public String unlock(String username, String owner,String filepath,int row,int column);
    public String[] getlockinfo(String username, String owner, String filepath);
}
